### PR TITLE
Prevent goroutine leak in oidc client

### LINKED ISSUE #11950: OIDC goroutine leak
client.SyncProviderConfig also returns a `stop` chan.  Don't we need to stop the background goroutine ? 
```go
stop = client.SyncProviderConfig(conn.GetIssuerURL())
defer func() {close(stop)} 
```

_Originally posted by @smallinsky in https://github.com/gravitational/teleport/pull/11944#discussion_r850570999_

This should be fixed in:
- master
- v9
- v8
- v7

### PR DESCRIPTION
**Pull request by rosstimothy**
Adds the stop channel returned from client.SyncProviderConfig to
the oidcClient and closes the channel when the client is cleaned
up. This causes the sync goroutine being run by go-oidc to end.

Fixes #11950

### CODE REVIEW

File: lib/auth/oidc.go [hunk 1]
```diff
@@ -79,14 +80,16 @@ func (a *Server) createOIDCClient(conn types.OIDCConnector) (*oidc.Client, error
 		return nil, trace.Wrap(err)
 	}
 
-	doneSyncing := make(chan struct{})
+	doneSyncing := make(chan chan struct{})
```
Comment (zmb3) [inline lib/auth/oidc.go:83]: Not every day you see a `chan chan`! 😅

File: lib/auth/oidc.go [hunk 2]
```diff
@@ -79,26 +80,37 @@ func (a *Server) createOIDCClient(conn types.OIDCConnector) (*oidc.Client, error
 		return nil, trace.Wrap(err)
 	}
 
-	doneSyncing := make(chan struct{})
+	// SyncProviderConfig doesn't take a context for cancellation, instead it
+	// returns a channel that has to be closed to stop the sync. So what we
+	// can do until it is changed to take a context, is to wait for the syncContext
```
Comment (codingllama) [inline lib/auth/oidc.go:85]: > can do until it is changed to take a context

Is this something you are planning to do? Let's simplify the comment if not.

**Review by codingllama — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

**Review by lxea — APPROVED** [review]

**Comment by zmb3** [comment]
@rosstimothy think we can get this merged and backported by EOD tomorrow? It would be great if we can include this in 9.1.0.

**Comment by rosstimothy** [comment]
> @rosstimothy think we can get this merged and backported by EOD tomorrow? It would be great if we can include this in 9.1.0.

I will do my best to get it in by then. The CI gods may have other ideas

**Comment by rosstimothy** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|branch/v7|[](https://github.com/gravitational/teleport/pull/12076)|
|✅|branch/v8|[](https://github.com/gravitational/teleport/pull/12077)|
|✅|branch/v9|[](https://github.com/gravitational/teleport/pull/12078)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**